### DISCUSSION TITLE
Can i scope the loading state of AsyncNotifier

### DISCUSSION THREAD

**Discussion by Velkamhell97** [rrousselGit/riverpod#2287] [General]
In the usage example of `AsyncNotifier` in `riverpod` docs https://docs-v2.riverpod.dev/docs/providers/notifier_provider, whenever i add a `TODO` to the list, the whole list widget is replaced by the loading widget, i want to know if there is some way to scope the loading state, for example when i add product only watch the loading state in the add button and keep the list with the last data state. Maybe i must use two separateds `AsyncNotifier` ?

Thanks for this great package.

**Comment by AhmedLSayed9** [comment]
This is an upcoming feature https://github.com/rrousselGit/riverpod/issues/1660
For now, You've to use two separate providers yes

**Reply by Velkamhell97** [reply]
In the linked issue that you mentioned, @rrousselGit says that this can be done using code-generation, do you know how or where can i found an example about this ?

**Reply by AhmedLSayed9** [reply]
It's still WIP
